### PR TITLE
Add "proto" to sourceSets for customizing proto file locations

### LINKED ISSUE #4: Integrate protoSources into sourceSets
A more Gradlely alternative to ``protoSources`` convention:
```groovy
sourceSets {
  main {
    java {
      srcDir 'src/java'
    }
    proto {
      srcDir 'src/proto'
    }
  }
}
```

Take a look at [GroovyBasePlugin](https://github.com/gradle/gradle/blob/eb4d045f12dab4fa12a7da7ee99b6a70fe024a66/subprojects/plugins/src/main/groovy/org/gradle/api/plugins/GroovyBasePlugin.java#L83)

### PR DESCRIPTION
**Pull request by zhangkun83**
Resolves #4, an alternative solution to #1

@ejona86 please review

### CODE REVIEW

File: src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy [hunk 1]
```diff
@@ -170,13 +196,20 @@ class ProtobufPlugin implements Plugin<Project> {
         project.tasks.create(generateJavaTaskName, ProtobufCompile) {
             description = "Compiles Proto source '${sourceSet.name}:proto'"
             def List<?> protoSources = new ArrayList<?>()
-            protoSources.add project.fileTree("src/${sourceSet.name}/proto") {include "**/*.proto"}
             protoSources.add project.fileTree("${project.extractedProtosDir}/${sourceSet.name}") {include "**/*.proto"}
             protoSources.addAll(project.convention.plugins.protobuf.protoSources.get(sourceSet.name))
```
Comment (ejona86) [inline src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy]: Is this line still necessary?

Comment (zhangkun83) [inline src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy]: Removed protoSources from convention.

File: src/main/groovy/com/google/protobuf/gradle/ProtobufConvention.groovy [hunk 2]
```diff
@@ -52,22 +52,6 @@ class ProtobufConvention {
     private def Multimap<String, ?> protoSources = new ArrayListMultimap()
```
Comment (ejona86) [inline src/main/groovy/com/google/protobuf/gradle/ProtobufConvention.groovy]: Remove this too?

**Comment by ejona86** [comment]
@zhangkun83, LGTM

File: src/main/groovy/com/google/protobuf/gradle/ProtobufConvention.groovy (hunk 2 above)
Comment (zhangkun83) [inline src/main/groovy/com/google/protobuf/gradle/ProtobufConvention.groovy]: Removed.

**Pull request merged**